### DISCUSSION TITLE
(FAST.Farm)Some questions about high-res domain precursor files in VTK format of the downstream wind turbine.

### DISCUSSION THREAD

**Discussion by Zhiming-Cai** [OpenFAST/openfast#1105] [Q&A] [answered]
Dear all:
      We know that the low-res domain in FAST.Farm is primarily responsible for wake meandering and merging; and the high-res wind domain in FAST.Farm is primarily responsible for ambient and waked inflow local to a turbine. Each of these ambient wind data files must follow the simple legacy serial VTK file format. And we can generate inflow vtk files for both domains using SOWFA. 
      However,  if there are two wind turbines in the wind field, the inflow of the downstream turbine is the wake of the upstream one, so how should the high-res domain of the downstream turbine be generated? I think it should contain the wake of the upstream wind turbine, but the wake is calculated by FAST.Farm. So how should I get the high-res domain of the downstream turbine?

@NRELAdmin @jjonkman @bjonkman

**Comment by jjonkman** [comment, marked as answer]
Dear @Helloczm,

If you are using a high-fidelity precursor (e.g., using SOWFA) to generate the VTK inflow to FAST.Farm (`Mod_AmbWind` = 1), the precursor simulation only needs to consider the ambient inflow conditions, and not include wind turbines or wakes.  The precursor should be generated such that both the low- and high-resolution domains needed by FAST.Farm can be extracted, e.g., with refinement zones for the high-resolution domains embedded where the turbines will be placed in FAST.Farm or with a fine resolution throughout.

You can also generate inflow for FAST.Farm using synthetic models such as TurbSim (`Mod_AmbWind` = 2 or 3).

FAST.Farm will model the wind turbines (in OpenFAST) and calculate the wake effects internally.

For more information, see the Inflow Wind Generation section of the online FAST.Farm User's Guide and Theory Manual on OpenFAST readthedocs: https://openfast.readthedocs.io/en/main/source/user/fast.farm/ModelGuidance.html#inflow-wind-generation.

Best regards,

**Comment by Zhiming-Cai** [comment]
Dear @jjonkman ,

Thanks for your answer! By the way, I'd like to consult you how to generate the high-res domain precursor files in VTK format by using SOWFA or TurbSim.

Best regards,

**Reply by jjonkman** [reply]
Dear @Helloczm,

I don't have more to add beyond my prior response.  More information is provided in the Inflow Wind Generation section of the online FAST.Farm User's Guide and Theory Manual on OpenFAST readthedocs: https://openfast.readthedocs.io/en/main/source/user/fast.farm/ModelGuidance.html#inflow-wind-generation.

Please note that TurbSim does not generate VTK files.

Best regards,
